### PR TITLE
ENT-4136: Fixed memory leaks in unit tests

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1640/)

https://ci.cfengine.com/job/pr-pipeline/1640/

File: libutils/sequence.c [hunk 1]
```diff
@@ -553,6 +553,7 @@ Seq *SeqStringDeserialize(const char *const serialized)
             || NULL == (new_str = ValidDuplicate(src, length))
             || src[length] != '\n')
         {
+            free(new_str);
```
Comment (karlhto) [inline libutils/sequence.c]: This can result in an invalid free if any of the first three if's hit.

Comment (olehermanse) [reply libutils/sequence.c]: @karlhto no, look further up at the diff. I initialized it to NULL. `free(NULL)` is valid.

Comment (olehermanse) [reply libutils/sequence.c]: From Linux man page:

> If ptr is NULL, no operation is performed.

From C99 standard:

> If ptr is a null pointer, no action occurs.

Comment (karlhto) [reply libutils/sequence.c]: Alright :+1:

**Pull request merged**